server/platform/services/cache/provider.go

```diff
@@ -40,28 +50,81 @@ func NewProvider() Provider {
 // NewCache creates a new cache with given opts
 func (c *cacheProvider) NewCache(opts *CacheOptions) (Cache, error) {
 	if opts.Striped {
-		return NewLRUStriped(LRUOptions{
-			Name:                   opts.Name,
-			Size:                   opts.Size,
-			DefaultExpiry:          opts.DefaultExpiry,
-			InvalidateClusterEvent: opts.InvalidateClusterEvent,
-			StripedBuckets:         opts.StripedBuckets,
-		})
+		return NewLRUStriped(opts)
 	}
-	return NewLRU(LRUOptions{
-		Name:                   opts.Name,
-		Size:                   opts.Size,
-		DefaultExpiry:          opts.DefaultExpiry,
-		InvalidateClusterEvent: opts.InvalidateClusterEvent,
-	}), nil
+	return NewLRU(opts), nil
 }
 
 // Connect opens a new connection to the cache using specific provider parameters.
-func (c *cacheProvider) Connect() error {
-	return nil
+func (c *cacheProvider) Connect() (string, error) {
+	return "OK", nil
+}
+
+func (c *cacheProvider) SetMetrics(metrics einterfaces.MetricsInterface) {
 }
 
 // Close releases any resources used by the cache provider.
 func (c *cacheProvider) Close() error {
 	return nil
 }
+
+func (c *cacheProvider) Type() string {
+	return model.CacheTypeLRU
+}
+
+type redisProvider struct {
+	client  rueidis.Client
+	metrics einterfaces.MetricsInterface
+}
+
+type RedisOptions struct {
+	RedisAddr     string
+	RedisPassword string
+	RedisDB       int
+}
+
+// NewProvider creates a new CacheProvider
+func NewRedisProvider(opts *RedisOptions) (Provider, error) {
+	client, err := rueidis.NewClient(rueidis.ClientOption{
+		InitAddress:       []string{opts.RedisAddr},
+		Password:          opts.RedisPassword,
+		SelectDB:          opts.RedisDB,
+		ForceSingleClient: true,
+		CacheSizeEachConn: 16 * (1 << 20), // 16MiB local cache size
```

Comment:
Do we have an idea of how many connections will be used? Is there a hard cap or something?